### PR TITLE
[BH-1333] CW2015 driver reactor

### PR DESCRIPTION
**Pull request by swichu91**
CW2015 fuel gauge refactor.
Fixed init logic.

### CODE REVIEW

**Review by wojtekrzepecki — APPROVED** [review]
Looks good, no big issues found

File: module-bsp/board/rt1051/bellpx/hal/battery_charger/BatteryCharger.cpp [hunk 1]
```diff
@@ -64,6 +64,7 @@ namespace hal::battery
         void sendNotification(Events event);
         SOC fetchBatterySOC() const;
         void pollFuelGauge();
+        bool try_enable_charging();
```
Comment (mjan-dev) [inline module-bsp/board/rt1051/bellpx/hal/battery_charger/BatteryCharger.cpp:67]: why the naming uses different convention? not tryEnableCharging

Comment (swichu91) [reply module-bsp/board/rt1051/bellpx/hal/battery_charger/BatteryCharger.cpp:67]: Good point! Fixed

**Review by mjan-dev — APPROVED** [review]
LGTM

**Review by paweljonskim — APPROVED** [review]

**Pull request merged**